### PR TITLE
Check whether component supports children for wrapping

### PR DESCRIPTION
**Pull request by bkrmendy**
## Problem
In the Wrap menu, we offer to wrap in elements that don't actually support children.

## Fix
The fix consist of two parts:
- Add a `supportsChildren` prop to `ComponentDescriptor`, supply this `supportsChildren` prop wherever possible, and filter the component list in the popup based on that.
- If there's no available `ComponentDescriptor`, this PR adds a best-effort check that tries to look up a top-level element with the same name as the one from the export, and checks whether that element uses the children prop

### CODE REVIEW

**Review by ruggi — APPROVED** [review]

File: editor/src/components/shared/project-components.ts [hunk 1]
```diff
@@ -283,12 +293,14 @@ function makeHTMLDescriptor(
   extraPropertyControls: PropertyControls,
   attributes?: () => JSXAttributes,
 ): ComponentDescriptor {
+  const supportsChildren = intrinsicHTMLElementNamesThatSupportChildren.includes(tag) // TODO: maybe circ dep
```
[the remark this answers is not in the document]
Comment (bkrmendy) [reply editor/src/components/shared/project-components.ts]: I removed the comment in https://github.com/concrete-utopia/utopia/pull/4429/commits/9dccb4aff90c77b6b3c110dbb387f5a32cad18f7, it seems not to cause any problems (famous last words)

**Review by seanparsons — APPROVED** [review]

File: editor/src/core/shared/project-file-types.ts [hunk 2]
```diff
@@ -761,3 +765,14 @@ export type NodeModules = {
...
+export function getTopLevelElementByExportsDetail(
+  file: ParseSuccess,
+  nameToLookFor: string,
+): UtopiaJSXComponent | null {
+  return (
+    file.topLevelElements.find(
+      (t): t is UtopiaJSXComponent => t.type === 'UTOPIA_JSX_COMPONENT' && t.name === nameToLookFor,
+    ) ?? null
+  )
+}
```
Comment (Rheeseyb) [inline editor/src/core/shared/project-file-types.ts:778]: Can you move this to `/editor/src/core/model/project-file-utils.ts` please, as we have a few functions close to this in there

File: editor/src/core/third-party/remix-controls.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ export const RemixRunReactComponents: ComponentDescriptorsForFile = {
     properties: {
       to: { control: 'string-input' },
     },
+    supportsChildren: false,
```
Comment (Rheeseyb) [inline editor/src/core/third-party/remix-controls.ts:15]:
```suggestion
    supportsChildren: true,
```

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/core/shared/project-file-types.ts (hunk 2 above)
Comment (bkrmendy) [reply editor/src/core/shared/project-file-types.ts]: moved it in https://github.com/concrete-utopia/utopia/pull/4429/commits/71c5121c27040496610686db54e94c105d6c59cb

**Pull request merged**